packages/entities/entities-plugins/src/components/free-form/Datakit/flow-editor/side-panel/CacheField.vue

```diff
@@ -0,0 +1,267 @@
+          class="dk-cache-inline-redis"
+        >
+          <div>{{ t('plugins.free-form.datakit.flow_editor.panel_segments.resources.cache.inline_config_preview') }}</div>
+          <KCodeBlock
+            id="dk-cache-redis-configuration"
+            :code="redisConfigYaml"
+            language="yaml"
+            theme="dark"
+            @code-block-render="highlight"
+          />
+        </div>
+      </div>
+    </KModal>
+  </div>
+</template>
+
+<script setup lang="ts">
+import composables from '../../../../../composables'
+import { AddIcon, DeployIcon, MoreIcon } from '@kong/icons'
+import type { CacheConfigFormData } from '../../types'
+import Field from '../../../shared/Field.vue'
+import RadioField from '../../../shared/RadioField.vue'
+import { computed, ref } from 'vue'
+import { useField, useFormShared } from '../../../shared/composables'
+import {
+  KUI_COLOR_TEXT_NEUTRAL,
+  KUI_COLOR_TEXT,
+} from '@kong/design-tokens'
+import {
+  RedisConfigurationSelector,
+} from '@kong-ui-public/entities-redis-configurations'
+import yaml from 'js-yaml'
+import { createSingletonShorthands, createdBundledHighlighter } from '@shikijs/core'
+import { createJavaScriptRegexEngine } from '@shikijs/engine-javascript'
+import type { CodeBlockEventData } from '@kong/kongponents'
+
+export type FormData = { cache?: CacheConfigFormData | null }
+
+defineProps<{
+  strategy?: CacheConfigFormData['strategy']
+}>()
+
+const emit = defineEmits<{
+  'update': [config: FormData]
+  'remove': []
+  'cancel': []
+}>()
+
+const { getDefault, formData } = useFormShared<FormData>()
+const field = useField<FormData['cache']>('cache')
+
+if (field.error) throw new Error('No cache schema')
+
+const modalVisible = ref(false)
+const { i18n: { t } } = composables.useI18n()
+const canSubmit = computed(() => {
+  if (localStrategy.value === 'redis' && !formData.cache!.partial_id && !formData.cache!.redis) {
+    return false
+  }
+  return true
+})
+
+const localStrategy = computed(() => field.value.value?.strategy)
+
+const redisConfigYaml = computed(() => {
+  if (!field.value.value?.redis) return ''
+  return yaml.dump(field.value.value.redis, {
+    indent: 2,
+    noRefs: true,
+  })
+})
+
+const handleOpen = () => {
+  modalVisible.value = true
+}
+
+const handleClose = () => {
+  modalVisible.value = false
+  emit('cancel')
+}
+
+const handleRemove = () => {
+  emit('remove')
+}
+
+const handleAdd = () => {
+  handleOpen()
+  if (!field.value.value) {
+    field.value.value = getDefault('cache')
+    // Set default strategy to memory
+    field.value.value!.strategy = 'memory'
+  }
+}
+
+const handleSave = () => {
+  emit('update', formData)
+  handleClose()
+}
+
+const createHighlighter = createdBundledHighlighter({
+  langs: {
+    yaml: () => import('@shikijs/langs/yaml'),
+  },
+  themes: {
+    'github-dark': () => import('@shikijs/themes/github-dark'),
+  },
+  engine: () => createJavaScriptRegexEngine(),
+})
```

Comment:
/cc @kingyue737 maybe we can use a unified version of this one here? Not sure if this was introduced to `public-ui` yet.